packages/react-conformance/src/types.ts

```diff
@@ -0,0 +1,3 @@
+export interface Details {
```

Comment:
Couple comments here:
- Instead of Details probably call this something like `IsConformantOptions` (open to other suggestions too). 
- constructorName should be required.
- Think about what other info should be included in this and go ahead and add props for them (which won't be used yet):
  - A way to indicate enabled/disabled tests
  - A way to pass in extra tests
  - Depending on what all we can cover with lint rules, we may also need:
    - `exportedAtTopLevel?: boolean` (default true) like in the old isConformant
      - They'll also need to provide the top-level export for the package, typed as `any`
    - The path to the component file

We also still haven't totally determined whether these tests should run automatically on a glob of component files or whether they should be manually called for each component. Or we could potentially have both options--one exported function which runs tests for an individual component, and one which runs them for a glob (with a larger and slightly different options object).